### PR TITLE
Use overloaded function to handle different strerror_r variants

### PR DESCRIPTION
**Pull request by joshuakraemer**
Closes: https://github.com/dosbox-staging/dosbox-staging/issues/1047

### CODE REVIEW

File: src/misc/support.cpp [hunk 1]
```diff
@@ -313,22 +313,19 @@ void E_Exit(const char *format, ...)
 	throw(e_exit_buf);
 }
 
+/* Overloaded function to handle different return types of POSIX and GNU strerror_r variants */
+const char *strerror_result(int retval, const char *buf) {return retval == 0 ? buf : NULL;}
+const char *strerror_result(const char *retval, const char *buf) {return retval;}
+
```
Comment (kcgen) [inline src/misc/support.cpp:319]: Thanks for this PR!
Can you adjust `NULL` --> `nullptr`
and then in your source tree, run `./scripts/format-commit.sh`.

Comment (joshuakraemer) [reply src/misc/support.cpp]: Done. :)

Comment (kcgen) [reply src/misc/support.cpp]: Inbound

**Comment by kcgen** [comment]
(PR merged locally at https://github.com/dosbox-staging/dosbox-staging/pull/1049).  Thank you @joshuakraemer!

**Pull request closed without merging**